docs/specs/sequences/endpoint_reconfiguration.md

```diff
@@ -0,0 +1,142 @@
+[**UDMI**](../../../) / [**Docs**](../../) / [**Specs**](../) / [**Sequences**](./) / [Endpoint Reconfiguration](#)
+
+# Endpoint Reconfiguration
+
+Endpoint reconfiguration is the reconfiguration of the UDMI endpoint through the
+UDMI blob delivery mechanisms via UDMI config messages.
+
+The [endpoint configuration blob](https://github.com/faucetsdn/udmi/blob/master/tests/configuration_endpoint.tests/simple.json) is a JSON object defined by
+[configuration_endpoint.json](https://faucetsdn.github.io/udmi/gencode/docs/configuration_endpoint.html), which is base64 encoded in the config message.
+
+
+## Tests
+
+### Valid Endpoint (Successful) Reconfiguration
+
+Note - `<ENDPOINT>` is a **base64** encoded endpoint object
+
+```mermaid
+%%{wrap}%%
+sequenceDiagram
+    autonumber
+    participant D as Device
+    participant E as Original Endpoint
+    participant E' as New Endpoint
+    E->>D:CONFIG MESSAGE<br>blobset.blobs._iot_endpoint_config.base64 = <ENDPOINT><br>blobset.blobs._iot_endpoint.phase = "final"
+    D->>E:STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.phase = "apply"
+    D-->>E':CONNECTION ATTEMPT
+    E'->>D:CONFIG MESSAGE
+    D->>E':STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.phase = "final"
+```
+
+### Invalid Endpoint (Unsuccessful Reconfiguration)
+
+```mermaid
+%%{wrap}%%
+sequenceDiagram
+    autonumber
+    participant D as Device
+    participant E as Original Endpoint
+    participant E' as New Endpoint
+    E->>D:CONFIG MESSAGE<br>blobset.blobs._iot_endpoint_config.blob = <ENDPOINT><br>blobset.blobs._iot_endpoint.blob.phase = "final"
+    D->>E:STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.phase = "apply"
+    D-->>E':CONNECTION ATTEMPT
+    note over D: Failure, e.g. endpoint doesn't exist, incorrect credentials, ...
+    D-->>E:CONNECTION ATTEMPT
+    E->>D:CONFIG MESSAGE
+    D->>E:STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.phase = "final"<BR/>blobset.blobs._iot_endpoint_config.status.level=500 (ERROR) 
```

Comment:
somewhere we need to introduce a category(s) for this too...

Reply:
Do we differentiate for different types of failure? Or is it a single `endpoint.reconfig.failed` or similar? 
A pretty common error I've encountered before is an endpoint "unreachable" because of firewall or network restrictions. Would that be its own category, or is that level of detail left to the message?

Reply:
Category added